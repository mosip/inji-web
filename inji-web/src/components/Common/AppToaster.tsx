import React from "react";
import {ToastContainer} from "react-toastify";

export const AppToaster: React.FC = () => {
    return <ToastContainer
        position="top-right"
        autoClose={1000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        icon={<React.Fragment/>}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        style={{width: '400px'}}
        theme="colored"
    />
}
