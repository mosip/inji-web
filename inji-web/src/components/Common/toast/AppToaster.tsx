import React from "react";
import {ToastContainer} from "react-toastify";
import {useSelector} from "react-redux";
import {RootState} from "../../../types/redux";
import {isRTL} from "../../../utils/i18n";

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    return (
        <div data-testid="Apptoaster-outer-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={3000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment/>}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                theme="colored"
            />
        </div>
    );
};
