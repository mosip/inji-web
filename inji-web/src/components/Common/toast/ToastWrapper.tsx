import React from "react";
import {toast, ToastOptions} from "react-toastify";

type ToastType = "info" | "success" | "warning" | "error" | "default";

interface ToastWrapperProps {
    message: string;
    type?: ToastType;
    testId?: string;
    options?: ToastOptions;
}

const ToastContent: React.FC<{ message: string; testId?: string }> = ({message, testId}) => (
    <div data-testid={testId}>{message}</div>
);

export const showToast = ({message, type = "default", testId, options}: ToastWrapperProps) => {
    const content = <ToastContent message={message} testId={`toast-${type}-${testId}`}/>;

    switch (type) {
        case "success":
            toast.success(content, options);
            break;
        case "error":
            toast.error(content, options);
            break;
        case "warning":
            toast.warning(content, options);
            break;
        case "info":
            toast.info(content, options);
            break;
        default:
            toast(content, options);
            break;
    }
};
