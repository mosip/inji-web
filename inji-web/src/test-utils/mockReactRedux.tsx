jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    return {
      ...ActualReactRedux,
      useSelector: jest.fn(),
      useDispatch: jest.fn(),
    };
  });
  
  // Helper to set the mock implementation of useSelector:
  export const setMockUseSelectorState = (state: any) => {
    const { useSelector } = require('react-redux') as { useSelector: jest.Mock };
    useSelector.mockImplementation((selector: any) => selector(state));
  };
  
  export const setMockUseDispatchReturnValue = (dispatchFn: jest.Mock) => {
    const { useDispatch } = require('react-redux') as { useDispatch: jest.Mock };
    (useDispatch as jest.Mock).mockReturnValue(dispatchFn);
};