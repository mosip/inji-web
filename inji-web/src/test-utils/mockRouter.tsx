export const mockNavigateFn = jest.fn();

jest.mock('react-router-dom', () => {
  const Actual = jest.requireActual('react-router-dom');
  return {
    ...Actual,
    useNavigate: () => mockNavigateFn,
    useLocation: jest.fn().mockReturnValue({ pathname: '/' }),
    Outlet: () => <div data-testid="Outlet">Page content</div>,
  };
});

export const setMockUseLocation = (locationObj: { pathname: string }) => {
  const { useLocation } = require('react-router-dom') as { useLocation: jest.Mock };
  useLocation.mockReturnValue(locationObj);
};
