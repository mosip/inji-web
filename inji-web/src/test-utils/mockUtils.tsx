import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

// Mock for storage module
export const mockStorageModule = () => {
    jest.mock('../utils/storage.ts', () => ({
        storage: {
            getItem: jest.fn(),
            setItem: jest.fn(),
            SESSION_INFO: 'SESSION_INFO',
            SELECTED_LANGUAGE: 'selectedLanguage'
        }
    }));
};

// Mock for localStorage
export const mockLocalStorage = () => {
    let store: { [key: string]: string } = {};
  
    const localStorageMock = {
      getItem: jest.fn((key: string) => store[key] || null),
      setItem: jest.fn((key: string, value: string) => {
        store[key] = value;
      })
    };
  
    Object.defineProperty(window, 'localStorage', { 
      value: localStorageMock,
      writable: true
    });
  
    return localStorageMock;
  };

// API Mock
export const mockApi = {
    authorizationRedirectionUrl: 'http://mockurl.com'
};

// Crypto Mock
export const mockCrypto = {
    getRandomValues: (array: Uint32Array) => {
        for (let i = 0; i < array.length; i++) {
            array[i] = 4294967296;
        }
        return array;
    },
    subtle: {} as SubtleCrypto,
    randomUUID: () => '123e4567-e89b-12d3-a456-426614174000'
} as Crypto;

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
 };

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};
              
export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};
    
export const mockUseParams = ()=>{
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({ issuerId: 'test-issuer-id' }),
      }));   
};

export const mockUseapiObject = () =>{
    jest.mock('../utils/api', () => ({
        api: {
          mimotoHost: 'http://mocked-api-host',
        },
      }));
};

export const mockUseFetch = () =>{
    jest.mock('../hooks/useFetch', () => {
        const RequestStatus = {
          LOADING: 'LOADING',
          DONE: 'DONE',
          ERROR: 'ERROR',
        };
        return {
          useFetch: () => jest.fn(),
          RequestStatus,
        };
      });
};

export const mockUseToast = () =>{
    jest.mock('react-toastify', () => ({
      toast: {
        error: jest.fn(),
      },
    }));
};

export const mockusemisc = ()=>{
    jest.mock('../utils/misc', () => ({
        downloadCredentialPDF: jest.fn(),
        getErrorObject: jest.fn(),
        getTokenRequestBody: jest.fn(),
      }));
};

export const mockusei18n = ()=>{
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key: string) => key,
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

};

export const mockWindowLocation = (url: string) => {
    const location = new URL(url) as unknown as Location;
    Object.defineProperty(window, 'location', {
      value: location,
      writable: true,
    });
  };

export const renderWithRouter = (Element: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
      <BrowserRouter>
        <Provider store={reduxStore}>
          <Routes>
            <Route path="*" element={Element} />
          </Routes>
        </Provider>
      </BrowserRouter>
    );
  };

