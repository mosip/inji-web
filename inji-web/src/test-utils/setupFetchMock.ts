export const fetchMock = jest.fn();

beforeAll(() => {
    global.fetch = fetchMock;
});

beforeEach(() => {
    fetchMock.mockReset();
});
