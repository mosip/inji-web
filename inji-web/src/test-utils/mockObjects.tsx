import { IssuerConfigurationObject } from '../types/data';
import { LandingPageWrapperProps } from '../components/Common/LandingPageWrapper';
import { NavBarProps } from '../types/components';
export const mockIssuerDisplayArrayObject = {
    name: "Name",
    language: "en",
    locale: "en",
    logo: {
        url: "https://url.com",
        alt_text: "alt text of the url"
    },
    title: "Title",
    description: "Description",
};

export const mockCredentialTypeDisplayArrayObject = {
    name: "Name",
    locale: "en",
    logo: "https://url.com",
};

export const mockCredentials: IssuerConfigurationObject = {
    credentials_supported: [
        {
            name: "InsuranceCredential",
            scope: "mosip_vc_ldp",
            display: [
                {
                    name: "Health Insurance",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        },
        {
            name: "AnotherCredential",
            scope: "mosip_ldp_vc",
            display: [
                {
                    name: "Another Credential",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        }
    ],
    "authorization_endpoint": "https://env.net/authorize",
    "grant_types_supported": ["authorization_code"]
};


export const mockIssuerObject = {
        name: 'Test Issuer',
        desc: 'Test Description',
        protocol: 'OTP' as 'OTP', // Explicitly set the type to 'OTP' or 'OpenId4VCI'
        issuer_id: 'test-issuer',
        authorization_endpoint: 'https://auth.test.com',
        credentials_endpoint: 'https://credentials.test.com',
        display: [{
            name: "Name",
            language: "en",
            locale: "en",
            logo: {
                url: "https://url.com",
                alt_text: "alt text of the url"
            },
            title: "Title",
            description: "Description",
        }],
        client_id: 'test-client-id',
        redirect_uri:'test-redirect-uri',
        token_endpoint:'test-token_endpoint',
        proxy_token_endpoint:'test-proxy_token_endpoint',
        client_alias:'',
        ovp_qr_enabled: true,
        scopes_supported: ['scope1', 'scope2']
}

export const mockIssuerObjectList =  [
    {
        name: 'Issuer 1',
        desc: 'Description 1',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer1',
        authorization_endpoint: 'https://issuer1.com/auth',
        credentials_endpoint: 'https://issuer1.com/credentials',
        display: [
            {
                name: 'Issuer 1',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer1.com/logo.png', alt_text: 'Issuer 1 Logo' },
                title: 'Issuer 1 Title',
                description: 'Issuer 1 Description',
            },
        ],
        client_id: 'client1',
        redirect_uri: 'https://issuer1.com/redirect',
        token_endpoint: 'https://issuer1.com/token',
        proxy_token_endpoint: 'https://issuer1.com/proxy-token',
        client_alias: 'client1-alias',
        ovp_qr_enabled: true,
        scopes_supported: ['openid', 'profile'],
    },
    {
        name: 'Issuer 2',
        desc: 'Description 2',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer2',
        authorization_endpoint: 'https://issuer2.com/auth',
        credentials_endpoint: 'https://issuer2.com/credentials',
        display: [
            {
                name: 'Issuer 2',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer2.com/logo.png', alt_text: 'Issuer 2 Logo' },
                title: 'Issuer 2 Title',
                description: 'Issuer 2 Description',
            },
        ],
        client_id: 'client2',
        redirect_uri: 'https://issuer2.com/redirect',
        token_endpoint: 'https://issuer2.com/token',
        proxy_token_endpoint: 'https://issuer2.com/proxy-token',
        client_alias: 'client2-alias',
        ovp_qr_enabled: false,
        scopes_supported: ['openid', 'profile'],
    },
]

export const mockLandingPageWrapperProps: LandingPageWrapperProps = {
    icon: <div data-testid="Test-Icon">Icon</div>,
    title: "Test Title",
    subTitle: "Test SubTitle",
    gotoHome: true,
};

export const mockNavBarProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};