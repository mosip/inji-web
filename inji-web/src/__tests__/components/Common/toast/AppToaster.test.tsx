import React from 'react';
import { AppToaster } from '../../../../components/Common/toast/AppToaster';
import { reduxStore } from '../../../../redux/reduxStore'; // Assuming the store is named reduxStore
import { renderWithProvider } from '../../../../test-utils/mockUtils';
import { toast } from 'react-toastify';

// Mock the useTranslation hook
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    initReactI18next: {
      type: '3rdParty',
      init: jest.fn(),
    },
  }));

// Mock the toast function
jest.mock('react-toastify', () => ({
  toast: jest.fn(),
  ToastContainer: jest.requireActual('react-toastify').ToastContainer,
}));

describe('Testing the Layout of AppToaster', () => {
  test('Check if the layout is matching with the snapshots for English language', () => {
    reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
    const { asFragment } = renderWithProvider(<AppToaster />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Check if the layout is matching with the snapshots for Arabic language', () => {
    reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
    const { asFragment } = renderWithProvider(<AppToaster />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Testing the Functionality of AppToaster', () => {
  test('Check if toast is called with the correct message for English language', () => {
    reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'en' });
    renderWithProvider(<AppToaster />);
    toast("AppToaster test message");
    expect(toast).toHaveBeenCalledWith("AppToaster test message");
  });

  test('Check if toast is called with the correct message for Arabic language', () => {
    reduxStore.dispatch({ type: 'SET_LANGUAGE', payload: 'ar' });
    renderWithProvider(<AppToaster />);
    toast("AppToaster test message");
    expect(toast).toHaveBeenCalledWith("AppToaster test message");
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
