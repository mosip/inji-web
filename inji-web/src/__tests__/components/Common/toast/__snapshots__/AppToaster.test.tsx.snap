// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for Arabic language 1`] = `
<DocumentFragment>
  <div
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of AppToaster Check if the layout is matching with the snapshots for English language 1`] = `
<DocumentFragment>
  <div
    data-testid="Apptoaster-outer-container"
  >
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
